git_machete/cmd.py

```diff
@@ -901,61 +901,71 @@ def print_new_line(new_status):
             print_new_line(True)
         if needs_slide_out:
             print_new_line(False)
-            ans = raw_input("Branch %s is merged into %s. Slide %s out of the tree of branch dependencies? [y/n/q] " %
+            ans = raw_input("Branch %s is merged into %s. Slide %s out of the tree of branch dependencies? [y/N/q/yq] " %
                             (bold(b), bold(u), bold(b))).lower()
-            if ans in ('y', 'yes'):
+            if ans in ('y', 'yes', 'yq'):
                 for d in down_branches.get(b) or []:
                     up_branch[d] = u
                 down_branches[u] = flat_map(lambda ud: (down_branches.get(b) or []) if ud == b else [ud],
                                             down_branches[u])
                 if b in annotations:
                     del annotations[b]
                 save_definition_file()
+                if ans == 'yq':
+                    raise StopTraversal
                 # No need to flush caches since nothing changed in commit/branch structure (only machete-specific changes happened).
-                continue  # No need to sync branch `b` with remote since it just got removed from the tree of dependencies.
+                continue  # No need to sync branch 'b' with remote since it just got removed from the tree of dependencies.
             elif ans in ('q', 'quit'):
-                return
+                raise StopTraversal
```

Comment:
It seems to be a change of logic. Could you split it into separate commit, please?

Reply:
I tidied up the raise/return logic in `traverse` and dependent functions, now return is used in `traverse` and `raise` in dependent functions, hope it's clearer now.